shap/plots/_image.py

```diff
@@ -127,3 +132,381 @@ def image(shap_values, pixel_values=None, labels=None, width=20, aspect=0.2, hsp
     cb.outline.set_visible(False)
     if show:
         pl.show()
+
+
+def image_to_text(shap_values):
+
+    from IPython.core.display import display, HTML
+
+    if len(shap_values.values.shape) == 5:
+        for i in range(shap_values.values.shape[0]):
+            display(HTML("<br/><b>"+ordinal_str(i)+" instance:</b><br/>"))
+            image_to_text(shap_values[i])
+        
+        return
+    
+
+    uuid = ''.join(random.choices(string.ascii_lowercase, k=20))
+    
+    # creating input html tokens
+    
+
+    model_output = shap_values.output_names
+
+    output_text_html = ''
+
+    for i in range(model_output.shape[0]):
+        output_text_html += "<div style='display:inline; text-align:center;'>" \
+                + f"<div id='{uuid}_output_flat_value_label_"+ str(i) +"'" \
+                + "style='display:none;color: #999; padding-top: 0px; font-size:12px;'>" \
+                + "</div>" \
+                + f"<div id='{uuid}_output_flat_token_"+ str(i) +"'" \
+                + "style='display: inline; background:transparent; border-radius: 3px; padding: 0px;cursor: default;cursor: pointer;'" \
+                + f"onmouseover=\"onMouseHoverFlat_{uuid}(this.id)\" " \
+                + f"onmouseout=\"onMouseOutFlat_{uuid}(this.id)\" " \
+                + f"onclick=\"onMouseClickFlat_{uuid}(this.id)\" " \
+                + ">" \
+                + model_output[i].replace("<", "&lt;").replace(">", "&gt;").replace(' ##', '').replace('▁', '') \
+                + " </div>" \
+                + "</div>"
+    
+    # computing gray scale images
+    
+    
+    image_data = shap_values.data
+    image_height = image_data.shape[0]
+    image_width = image_data.shape[1]
+    
+    
+    # computing gray scale image
+    
+    image_data_gray_scale = np.ones((image_height, image_width, 4)) * 255 * 0.5
```

Comment:
I see "4" used here when allocating np.ones, but only 0, 1, 2 are specified below (not 3rd axis), is 3rd axis supposed to be all 1's?

also, these two lines are the same, is that meant to be the case:
    image_data_gray_scale[:,:,1] = image_data_gray_scale[:,:,0]
    image_data_gray_scale[:,:,2] = image_data_gray_scale[:,:,0]

Reply:
Yes the 3rd axis represent alpha, its multiplied by 0.5 to make it look translucent, to allow for the shap overlay to be visible. 

Ans yes the two lines are meant to be the same, its just calculating average value of RGB and set it to all 2 channels (to make it grayscale).